From 59ad66a9f1d94e8fc6ef0cf34293ca7cd8ced590 Mon Sep 17 00:00:00 2001
From: Ales Nezbeda <anezbeda@redhat.com>
Date: Tue, 17 Sep 2024 09:17:24 +0200
Subject: [PATCH] Fix builds on EPEL9 due to outdated components

Some of the new functionalities that are present in Fedora and are
needed for distgen to build using the refreshed build system are not
present in EPEL9.

This means that we have to rollback some changes to allow build on
EPEL9. This shouldn't impact functionality of the application, but
keeping it in the main repository would create mess in the codebase.
It can also be unpredictable how will setuptools work with different
config files, so it's better to just use one config.
---
 setup.cfg | 36 ++++++++++++++++++++++++++++++++++++
 setup.py  | 22 ++++++++++++++++++++++
 2 files changed, 58 insertions(+)
 create mode 100644 setup.py

diff --git a/setup.cfg b/setup.cfg
index acd96a0..3151c89 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,8 +1,45 @@
+[metadata]
+name = distgen
+version = 2.0
+author = Red Hat Databases team
+description = Templating system/generator for distributions
+long_description = file: README.md
+license = GPLv2+
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Programming Language :: Python :: 3
+    License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)
+
+[options]
+setup_requires =
+    argparse-manpage[setuptools]
+include_package_data = True
+packages = find:
+python_requires = >=3.0
+
+[options.package_data]
+distgen = distconf/*.yaml, distconf/**/*.yaml, templates/*.tpl, templates/**/*.tpl, templates/**/**/*.tpl
+
+[options.packages.find]
+exclude =
+    rpm*
+    man*
+    build_manpages*
+
+[options.entry_points]
+console_scripts =
+    dg = distgen.__main__:main
+
+[build_manpages]
+manpages =
+    man/dg.1:object=parser:pyfile=distgen/cli_parser.py
+
 [flake8]
-ignore = 
-	E402,
-	C0116
-max-complexity = 15
+ignore=
+    # E402 module level import not at top of file
+    E402,
+    C0116
+max-complexity=15
 
 [egg_info]
 tag_build = 
diff --git a/setup.py b/setup.py
new file mode 100644
index 0000000..ac2b9de
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,17 @@
+from setuptools import setup
+
+from setuptools.command.build_py import build_py
+from setuptools.command.install import install
+
+from build_manpages.build_manpages import build_manpages
+from build_manpages import get_build_py_cmd
+from build_manpages import get_install_cmd
+
+setup(
+    cmdclass={
+        'build_manpages': build_manpages,
+        'build_py': get_build_py_cmd(build_py),
+        'install': get_install_cmd(install),
+    },
+)
+
-- 
2.46.0

